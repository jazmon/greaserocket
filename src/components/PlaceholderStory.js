// @flow
import React from 'react';
import styled from 'styled-components/native';
import PlaceholderComponent from 'components/PlaceholderComponent';

const Container = styled.View`
  height: 120;
  flex: 1;
  flex-direction: row;
  margin-bottom: 8;
  opacity: 0.5;
`;
const ImageContainer = styled.View`
  align-items: center;
  justify-content: center;
  width: 120;
`;
const MockImage = styled(PlaceholderComponent)`
  width: 100;
  height: 100;
`;
const TextArea = styled.View`
  flex: 1;
  padding: 16;
`;
const MockText = styled(PlaceholderComponent)`
  flex-grow: 0;
  flex-direction: row;
  height: 14;
  padding-top: ${props => (props.upper ? 2 : 0)};
  padding-bottom: 2;
  margin-top: ${props => (props.lower ? 2 : 0)};
`;

const PlaceholderStory = () => (
  <Container>
    <ImageContainer>
      <MockImage />
    </ImageContainer>
    <TextArea>
      <MockText upper />
      <MockText lower />
    </TextArea>
  </Container>
);
export default PlaceholderStory;
