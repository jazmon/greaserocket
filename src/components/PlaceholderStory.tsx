// @flow
import React from 'react';
import styled from 'styled-components/native';
import PlaceholderComponent from 'components/PlaceholderComponent';

const Container = styled.View`
  height: 120px;
  flex: 1;
  flex-direction: row;
  margin-bottom: 8px;
  opacity: 0.5;
`;
const ImageContainer = styled.View`
  align-items: center;
  justify-content: center;
  width: 120px;
`;
const MockImage = styled(PlaceholderComponent)`
  width: 100px;
  height: 100px;
`;
const TextArea = styled.View`
  flex: 1;
  padding: 16px;
`;

interface MockTextProps {
  upper?: boolean;
  lower?: boolean;
}
const MockText = styled(PlaceholderComponent)`
  flex-grow: 0;
  flex-direction: row;
  height: 14px;
  padding-top: ${(props: MockTextProps) => (props.upper ? 2 : 0)};
  padding-bottom: 2px;
  margin-top: ${(props: MockTextProps) => (props.lower ? 2 : 0)};
`;

const PlaceholderStory = () =>
  <Container>
    <ImageContainer>
      <MockImage />
    </ImageContainer>
    <TextArea>
      <MockText upper />
      <MockText lower />
    </TextArea>
  </Container>;
export default PlaceholderStory;
