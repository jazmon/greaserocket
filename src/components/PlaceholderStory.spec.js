import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import PlaceholderStory from './PlaceholderStory';


it('renders correctly', () => {
  const tree = renderer.create(
    <PlaceholderStory />
  );
  expect(tree).toMatchSnapshot();
});
