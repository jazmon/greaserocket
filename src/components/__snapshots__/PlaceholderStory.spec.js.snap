// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
        "height": 120,
        "marginBottom": 8,
        "opacity": 0.5,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "width": 120,
        },
        undefined,
      ]
    }
  >
    <BVLinearGradient
      colors={
        Array [
          4284900966,
          4288256409,
        ]
      }
      end={
        Object {
          "x": 1,
          "y": 0.5,
        }
      }
      locations={null}
      start={
        Object {
          "x": 0,
          "y": 0.5,
        }
      }
      style={
        Array [
          Object {},
          Object {
            "height": 100,
            "width": 100,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        },
        undefined,
      ]
    }
  >
    <BVLinearGradient
      colors={
        Array [
          4284900966,
          4288256409,
        ]
      }
      end={
        Object {
          "x": 1,
          "y": 0.5,
        }
      }
      locations={null}
      start={
        Object {
          "x": 0,
          "y": 0.5,
        }
      }
      style={
        Array [
          Object {},
          Object {
            "flexDirection": "row",
            "flexGrow": 0,
            "height": 14,
            "marginTop": 0,
            "paddingBottom": 2,
            "paddingTop": 2,
          },
        ]
      }
      upper={true}
    />
    <BVLinearGradient
      colors={
        Array [
          4284900966,
          4288256409,
        ]
      }
      end={
        Object {
          "x": 1,
          "y": 0.5,
        }
      }
      locations={null}
      lower={true}
      start={
        Object {
          "x": 0,
          "y": 0.5,
        }
      }
      style={
        Array [
          Object {},
          Object {
            "flexDirection": "row",
            "flexGrow": 0,
            "height": 14,
            "marginTop": 2,
            "paddingBottom": 2,
            "paddingTop": 0,
          },
        ]
      }
    />
  </View>
</View>
`;
